samples/bluetooth/bap_broadcast_sink/src/main.c

```diff
@@ -559,38 +519,49 @@ static int bis_sync_req_cb(struct bt_conn *conn,
 			   const struct bt_bap_scan_delegator_recv_state *recv_state,
 			   const uint32_t bis_sync_req[CONFIG_BT_BAP_BASS_MAX_SUBGROUPS])
 {
-	/* Bit field indicating from which subgroup(s) BIS sync is requested */
-	uint32_t requested_subgroup_sync = 0; /* currently only used for printout */
+	bool sync_req = false;
+	bool bis_sync_req_pref = false;
+	uint8_t bis_sync_req_cnt = 0;
+	uint32_t bis_sync_req_bitfield = 0;
 
-	requested_bis_sync = 0;
+	(void)memset(requested_bis_sync, 0, sizeof(requested_bis_sync));
 
 	for (uint8_t subgroup = 0; subgroup < recv_state->num_subgroups; subgroup++) {
+
+		printk("bis_sync_req[%u] = 0x%0x\n", subgroup, bis_sync_req[subgroup]);
 		if (bis_sync_req[subgroup] != 0) {
-			if (requested_bis_sync == 0) {
-				requested_bis_sync = bis_sync_req[subgroup];
-			} else {
-				if (requested_bis_sync != BT_BAP_BIS_SYNC_NO_PREF &&
-				    bis_sync_req[subgroup] != BT_BAP_BIS_SYNC_NO_PREF) {
-					/* Spec a little bit unclear. Here we choose to say that
-					 * BIS sync request from more than 1 subgroup is not
-					 * possible unless sync value is 0 or
-					 * BT_BAP_BIS_SYNC_NO_PREF
-					 */
-					printk("Unsupported BIS sync request from more than 1 "
-					       "subgroup\n");
-					return -EINVAL;
-				}
+			requested_bis_sync[subgroup] = bis_sync_req[subgroup];
+			if (bis_sync_req[subgroup] != BT_BAP_BIS_SYNC_NO_PREF) {
+				bis_sync_req_pref = true;
 			}
-			requested_subgroup_sync |= BIT(subgroup);
+			bis_sync_req_bitfield |= bis_sync_req[subgroup];
+			bis_sync_req_cnt++;
+			sync_req = true;
 		}
 	}
 
-	printk("BIS sync req for %p: BIS indexes 0x%08x (subgroup indexes 0x%08x), "
-	       "broadcast id: 0x%06x, (%s)\n",
-	       recv_state, requested_bis_sync, requested_subgroup_sync, recv_state->broadcast_id,
-	       big_synced ? "BIG synced" : "BIG not synced");
+	if (sync_req) {
+		if (bis_sync_req_pref) {
```

Comment:
Since `sync_req` is always true when `bis_sync_req_pref` we can just do

```suggestion
	if (bis_sync_req_pref) {
```

Reply:
Well spotted